lib/_http_server.js

```diff
@@ -635,11 +635,13 @@ function resOnFinish(req, res, socket, state, server) {
   process.nextTick(emitCloseNT, res);
 
   if (res._last) {
-    if (typeof socket.destroySoon === 'function') {
-      socket.destroySoon();
-    } else {
+    if (socket.writable)
       socket.end();
-    }
+
+    if (socket.writableFinished)
+      socket.destroy();
+    else
+      socket.once('finish', socket.destroy);
```

Comment:
Do we even need these calls to `destroy()` doesn't `autoDestroy` already handle this?